### ISSUE TITLE
Lighthouse SEO dosen't check accsessible name when checking "non-descriptive link name" 

### ISSUE THREAD

**Issue by MarieLowzowMorland** [GoogleChrome/lighthouse#12517]
From WCAG, link texts are covered in this article, that describes some of the motivation behind demanding descriptive links. https://www.w3.org/WAI/WCAG21/Understanding/link-purpose-in-context.html

One of the techniques mentioned here for passing is to use i.e aria-label to provide a good description in the cases where the author of the page for some reason wants to use a common repeating text such as "read more": https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA8 

However, Lighthouse has some code that blacklists common terms in link texts regardless of the what the accsessible name provided with i.e aria-label says, in stark contrast to the accsessible guidelines above: 
https://github.com/GoogleChrome/lighthouse/blob/545176a04fa51a68ce250a47ecbb9b9be1367c7c/lighthouse-core/audits/seo/link-text.js#L127

Shouldn't a SEO use the accsessible name over the text for anchor elements?
If so, then the lighthouse check should also use check for an accessible name before adding a penalty here

**Comment by paulirish** [comment]
I can empathize that this makes sense... however in a recent thread we heard that Google Search doesn't consider it: https://twitter.com/JohnMu/status/1400539761392164866

So, oddly.. this is Works As (Intentionally) Intended.
